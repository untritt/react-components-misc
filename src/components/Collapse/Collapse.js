import React from "react";
import cn from "classnames";

export default class Collapse extends React.Component {
  static defaultProps = {
    opened: false
  };

  state = { opened: this.props.opened };

  onClick = e => {
    e.preventDefault();
    this.setState({ opened: !this.state.opened });
  };

  render() {
    return (
      <div>
        <p>
          <a className="btn btn-primary" href="#" onClick={this.onClick}>
            Click me
          </a>
        </p>
        <div className={cn({ collapse: true, show: this.state.opened })}>
          <div className="card card-body">{this.props.text}</div>
        </div>
      </div>
    );
  }
}
