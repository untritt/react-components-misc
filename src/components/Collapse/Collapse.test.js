import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Collapse from "./Collapse";

Enzyme.configure({ adapter: new Adapter() });

test("Collapse", () => {
  const wrapper = mount(<Collapse text="hi" />);
  expect(wrapper.render()).toMatchSnapshot();

  const button = wrapper.find("a");

  button.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  button.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();

  button.simulate("click");
  expect(wrapper.render()).toMatchSnapshot();
});
