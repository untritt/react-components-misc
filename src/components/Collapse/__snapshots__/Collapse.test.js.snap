// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse 1`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Click me
    </a>
  </p>
  <div
    class="collapse"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;

exports[`Collapse 2`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Click me
    </a>
  </p>
  <div
    class="collapse show"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;

exports[`Collapse 3`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Click me
    </a>
  </p>
  <div
    class="collapse"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;

exports[`Collapse 4`] = `
<div>
  <p>
    <a
      class="btn btn-primary"
      href="#"
    >
      Click me
    </a>
  </p>
  <div
    class="collapse show"
  >
    <div
      class="card card-body"
    >
      hi
    </div>
  </div>
</div>
`;
